File: prime_factorisation.cs
using System;
using System.Collections.Generic;

class Program
{
    static List<int> PrimeFactorisation(int n)
    {
        List<int> factors = new List<int>();
        int divisor = 2;

        while (n > 1)
        {
            while (n % divisor == 0)
            {
                factors.Add(divisor);
                n /= divisor;
            }
            divisor++;
        }

        return factors;
    }
}

File: prime_factorisation.java
import java.util.ArrayList;
import java.util.List;

public class PrimeFactorisation {

    public static List<Integer> primeFactorisation(int n) {
        List<Integer> factors = new ArrayList<>();
        int divisor = 2;

        while (n > 1) {
            while (n % divisor == 0) {
                factors.add(divisor);
                n /= divisor;
            }
            divisor++;
        }

        return factors;
    }
}
